src/Test/StrictCheck/Shaped.hs

```diff
@@ -313,32 +313,34 @@ interleave p = unfold (fmap (project p)) . p
 -- @h % a@.
 unzipWith
   :: (All Functor [f, g, h], Shaped a)
-  => (forall x. f x -> (g x, h x))
+  => (forall x sx. sx ~ (Shape x ((%) g), Shape x ((%) h))
+       => f sx -> (g sx, h sx))
   -> (f % a -> (g % a, h % a))
 unzipWith split =
-  unPair . fold (crunch . pair . split)
+  unPair . fold (pair . bimap (Wrap . fmap fst) (Wrap . fmap snd)
+                 . split
+                 . fmap crunch)
 
 -- | The monadic equivalent of @unzipWith@; effectfully unzips an interleaved
 -- structure
 unzipWithM
   :: (Traversable f, All Functor [g, h], Shaped a, Monad m)
-  => (forall x. f x -> m (g x, h x))
+  => (forall x sx. sx ~ (Shape x ((%) g), Shape x ((%) h))
+       => f sx -> m (g sx, h sx))
   -> (f % a -> m (g % a, h % a))
 unzipWithM split =
-  fmap unPair . foldM (fmap (crunch . pair) . split)
+  fmap unPair . foldM (fmap (pair . bimap (Wrap . fmap fst) (Wrap . fmap snd))
+                       . split
+                       . fmap crunch)
 
 -- Some helpers for zipping and unzipping...
 
```

Comment:
I'm not sure `crunch` is the right name for this operation any more (we've changed what it does). I can't come up with a good verb for it, though...

Reply:
I didn't really understand the name in the first place, so I just left it.

Reply:
`unzipShape`?